theano/gpuarray/tests/test_dnn.py

```diff
@@ -2373,6 +2373,14 @@ def _meshgrid(height, width):
     img_out_gpu, = st_dnn_func(img, transform)
     img_out = np.asarray(img_out_gpu)
 
+    t_dy = T.tensor4('dy')
+    img_grad = T.grad(None, wrt=[t_img, t_theta], known_grads={st_dnn: t_dy})
+
+    grad_fn = theano.function([t_img, t_theta, t_dy], img_grad)
+
+    dy = -1 + 2 * np.random.randn(*img.shape).astype(theano.config.floatX)
+    spatialtf_grad = grad_fn(img, transform, dy)
```

Comment:
Strange.

But, to start, please apply my comments about `grad_undefined()`. Currently, `make_node()` returns 6 outputs, but C function (`dnn_sptf_grad()`) will generate only 2 outputs. `make_node()` must not return any `grad_undefined()`.

Let's first stabilize the code, and then we will check errors.